x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/whitelist/InternalEqlScriptUtils.java

```diff
@@ -33,6 +36,20 @@
     InternalEqlScriptUtils() {
     }
 
+    public static <T> Boolean multiValueDocValues(Map<String, ScriptDocValues<T>> doc, String fieldName, Predicate<T> script) {
+        ScriptDocValues<T> docValues = doc.get(fieldName);
+        if (docValues != null && docValues.isEmpty() == false) {
+            for (T value : docValues) {
+                if (script.test(value)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+        // missing value means "null"
+        return script.test(null);
```

Comment:
interesting, I hadn't really thought about passing `null` into the script. is it more correct to do that or to return `false` here? I'm not sure what expressions would cause this to evaluate true. `string(x != null) == "true"` is an interesting case

Reply:
I think your logic is correct. Interesting find!
I'll take a look at my implementation and see what I need to update.

Reply:
Evaluating missing as `null` is important exactly for the case you mention: comparison with `null`. If the script itself is returning `false` instead of `null` (for example `concat(field1, field2) != null`) these `null` operations wouldn't be possible.